src/cosmos/engine/CosmosEngine.ts

```diff
@@ -248,6 +283,97 @@ export class CosmosEngine extends CurrencyEngine<
             }
             return out
           }
+          case 'MakeTxDexSwap': {
+            const { fromTokenId, fromNativeAmount, metadata, txData } = params
+
+            if (txData == null) {
+              throw new Error('Cosmos MakeTxDexSwap requires txData')
+            }
+
+            const memos: EdgeMemo[] = []
+
+            let gasFeeCoin = coin('0', this.networkInfo.nativeDenom)
+            let gasLimit = '200000' // Default gas limit for swaps
+            let networkFee = '0'
+            let swapMemo: string | undefined
+
+            try {
+              const swapTxData = asCosmosSwapTransactionDataFromJson(txData)
+
+              // Extract memo - critical for cross-chain routing
+              if (swapTxData.memo != null && swapTxData.memo !== '') {
+                swapMemo = swapTxData.memo
+                memos.push({ type: 'text', value: swapMemo })
+                this.log(`DEX swap memo: ${swapMemo}`)
+              }
+
+              // Extract fee information from swap transaction data
+              if (swapTxData.fee.amount.length > 0) {
+                const feeAmount = swapTxData.fee.amount[0]
+                networkFee = feeAmount.amount
+                gasFeeCoin = coin(networkFee, feeAmount.denom)
+              }
+
+              gasLimit = swapTxData.fee.gas
+            } catch (e) {
+              // Fall back to defaults if parsing fails
```

Comment:
If the asCosmosSwapTransactionDataFromJson  cleaner throws here it's going to throw in signTx and therefore fail. It should fail here instead